src/lando/api/tests/test_git.py

```diff
@@ -0,0 +1,81 @@
+import subprocess
+
+import pytest
+
+from lando.api.tests.conftest import git_setup_user
+from lando.main.scm.git import GitSCM
+
+
+# We can't use fixtures directly in parametrize [0],
+# so we jumps through some hoops.
+# [0] https://github.com/pytest-dev/pytest/issues/349
+@pytest.mark.parametrize(
+    "path,repo_fixture_name,expected",
+    (
+        ("/non-existent-path", None, False),
+        ("/tmp", None, False),
+        (None, "git_repo", True),
+    ),
+)
+def test_is_initialised(path, repo_fixture_name, expected, request):
+    if not path:
+        path = request.getfixturevalue(repo_fixture_name).strpath
+    scm = GitSCM(path)
+    assert scm.repo_is_initialized == expected
+
+
+def test_clone(tmpdir, git_repo):
```

Comment:
This test could be a bit more robust. At least checking whether the correct `git` command was called, even with a mock.

Reply:
Ah, yep. I added a `_monkeypatch_scm` function that allows to mock `_git_run` but to retain the side effect, so we can test that
* The correct `git` commands are called
* Their side effects are actually as desired

(And this allowed me to spot a bug 😅)